boa/src/builtins/value/operations.rs

```diff
@@ -188,10 +173,8 @@ pub fn same_value(x: &Value, y: &Value, strict: bool) -> bool {
         return false;
     }
 
-    if x.get_type() == "number" {
-        let native_x = f64::from(x);
-        let native_y = f64::from(y);
-        return native_x.abs() - native_y.abs() == 0.0;
+    if x.is_number() {
```

Comment:
We should add a `// TODO` comment here and there to remember to check for `BigInt` in #358.